src/web.rs

```diff
@@ -48,3 +51,159 @@ impl WebWindowHandle {
         Self { id }
     }
 }
+
+/// Raw window handle for a Web canvas registered via [`wasm-bindgen`].
+///
+/// ## Construction
+/// ```no_run
+/// # use raw_window_handle::Wbg02CanvasWindowHandle;
+/// # use core::{ffi::c_void, ptr::NonNull};
+/// # fn get_canvas() -> NonNull<c_void> { unimplemented!() }
+/// let obj: NonNull<c_void> = get_canvas();
+/// let mut window_handle = Wbg02CanvasWindowHandle::new(obj);
+/// /* set fields */
+/// ```
+///
+/// [`wasm-bindgen`]: https://crates.io/crates/wasm-bindgen
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Wbg02CanvasWindowHandle {
+    /// The object representing the [`HtmlCanvasElement`].
+    ///
+    /// It is implied that this object is registered in the [`wasm-bindgen`] table and is an instance
+    /// of [`HtmlCanvasElement`]. The pointer is a direct reference to a [`JsValue`].
```

Comment:
Nit: wording ("it is implied" sounds odd).

```suggestion
    /// A pointer to the [`JsValue`] of a [`HtmlCanvasElement`].
    ///
    /// The object must be registered in the [`wasm-bindgen`] table.
```

(Unsure what you meant my "registered", so I kept it)